test/jdk/javax/swing/JFileChooser/6798062/bug6798062.java

```diff
@@ -21,73 +21,94 @@
  * questions.
  */
 
-/* @test %W% %E%
-   @bug 6798062
-   @summary Memory Leak on using getFiles of FileSystemView
-   @author Pavel Porvatov
-   @modules java.desktop/sun.awt
-            java.desktop/sun.awt.shell
-   @run applet/manual=done bug6798062.html
-*/
+/*
+ * @test
+ * @bug 6798062
+ * @requires (os.family == "windows")
+ * @summary Memory Leak on using getFiles of FileSystemView
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @modules java.desktop/sun.awt
+ *          java.desktop/sun.awt.shell
+ * @run main/manual bug6798062
+ */
 
 import sun.awt.OSInfo;
 import sun.awt.shell.ShellFolder;
 
-import javax.swing.*;
+import java.awt.Dimension;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
-import java.awt.*;
 import java.io.File;
 import java.io.FileNotFoundException;
-
-public class bug6798062 extends JApplet {
-
-    private final JSlider slider = new JSlider(0, 100);
-
-    private final JTextField tfLink = new JTextField();
-
-    private final JButton btnStart = new JButton("Start");
-
-    private final JButton btnStop = new JButton("Stop");
-
-    private final JButton btnGC = new JButton("Run System.gc()");
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JSlider;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+
+public class bug6798062 {
+
+    private static final String INSTRUCTIONS = """
+            The test is suitable only for Windows.
+
+            1. Create a link
+            2. Copy path to the link into TextField
+            3. Run the Windows Task Manager. Select the Processes tab and find the java process
+            4. Press the Start button in the test window
+            5. Wait several minutes and observe in the Windows Task Manager
+               that Memory Usage of java process is not increasing
+            If memory usage is increasing, click Fail else click Pass .  """;
+
+    private static JSlider slider;
+    private static JFrame frame;
+    private static JTextField tfLink;
```

Comment:
`tfLink` and `btnGC` can be initialized inside `initialize()` and moved there.

Reply:
As I understood, testUI calls the passed method in EDT so I think it's better to init the swing variables there..

Reply:
> As I understood, testUI calls the passed method in EDT so I think it's better to init the swing variables there..

Yes, that's true, if you pass a lambda expression to `testUI`, it is called on EDT.

Since `initialize` is called from `createUI`, both methods will be called on EDT.